File: src/weaponsprocurement/gui/WimGuiDialogOpener.java
package weaponsprocurement.gui;

import com.fs.starfarer.api.campaign.InteractionDialogAPI;

final class WimGuiDialogOpener {
    private WimGuiDialogOpener() {
    }

    static void show(InteractionDialogAPI dialog, float width, float height, WimGuiDialogPanel panel) {
        if (dialog == null) {
            throw new IllegalArgumentException("Cannot show a WP GUI dialog without an active interaction dialog.");
        }
        dialog.showCustomVisualDialog(width, height, new WimGuiDialogDelegate(panel));
    }
}

File: src/main/kotlin/weaponsprocurement/gui/WimGuiDialogOpener.kt
package weaponsprocurement.gui

import com.fs.starfarer.api.campaign.InteractionDialogAPI

class WimGuiDialogOpener private constructor() {
    companion object {
        @JvmStatic
        fun show(dialog: InteractionDialogAPI?, width: Float, height: Float, panel: WimGuiDialogPanel) {
            if (dialog == null) {
                throw IllegalArgumentException("Cannot show a WP GUI dialog without an active interaction dialog.")
            }
            dialog.showCustomVisualDialog(width, height, WimGuiDialogDelegate(panel))
        }
    }
}
